lib/battle/DamageCalculator.cpp

```diff
@@ -124,7 +127,22 @@ int DamageCalculator::getActorAttackBase() const
 
 int DamageCalculator::getActorAttackEffective() const
 {
-	return getActorAttackBase() + getActorAttackSlayer();
+	return getActorAttackBase() + getActorAttackSlayer() + getActorAttackIgnored();
+}
+
+int DamageCalculator::getActorAttackIgnored() const
+{
+	double multAttackReduction = battleBonusValue(info.defender, Selector::type()(BonusType::ENEMY_ATTACK_REDUCTION)) / 100.0;
+
+	if(multAttackReduction > 0)
+	{
+		int defaultRoundingMode = std::fegetround();
+		std::fesetround(FE_TOWARDZERO);
+		int reduction = std::nearbyint(multAttackReduction * getActorAttackBase());
+		std::fesetround(defaultRoundingMode);
```

Comment:
why not simply floor it?

Reply:
Only for this particular ability? Would be surprised to be honest.
For all rounding operations? In that case - this is wrong place for such change.

Reply:
i dont know if it works that way for this aprticular ability but i assume you checked this kind of stuff when you fixed behemoth damage etc. and I am not eager to check other abilities at the moment

Reply:
Nope. I did not.

Besides - these are float.
They *will* drift and return imprecise results and they *will* work differently on different platforms. So I'd rather keep code simple unless we know that this is correct implementation

Reply:
Or alternatively - use int-only version, e.g.
`int reduction = (getActorAttackBase() * multAttackPercentage + 49) / 100`
So compiler / CPU can not play with floating point rounding / precision